container/lxd/lxd.go

```diff
@@ -165,29 +166,18 @@ func (manager *containerManager) CreateContainer(
 		logger.Infof("instance %q configured with %v network devices", name, nics)
 	}
 
-	// Push the required /etc/network/interfaces file to the container.
-	// By pushing this file (which happens after LXD init, and before LXD
-	// start) we ensure that we get Juju's version of ENI, as opposed to
-	// the default LXD version, which may assume it can do DHCP over eth0.
-	// Especially on a multi-nic host, it is possible for MAAS to provide
-	// DHCP on a different space to that which the container eth0 interface
-	// will be bridged, or not provide DHCP at all.
-	eni, err := containerinit.GenerateNetworkConfig(networkConfig)
-	if err != nil {
-		err = errors.Annotatef(err, "failed to generate /etc/network/interfaces content")
-		return
-	}
-
 	spec := lxdclient.InstanceSpec{
 		Name:     name,
 		Image:    imageName,
 		Metadata: metadata,
 		Devices:  nics,
 		Profiles: profiles,
+		// We overwrite the (default) eth0.cfg file so that the trusty container
```

Comment:
This didn't seem to do what you have commented here. I still see on LXD the container coming up with "listening on LPF/eth0/..." and and then ifdown stopping that and doing DHCPRELEASE. I have the feeling that cloud-init is creating the file always on Trusty, so it doesn't help us to overwrite it. Have you tried without this and seen if it acts differently?